sources/daospaceplugin/WithdrawPlugin.move

```diff
@@ -0,0 +1,145 @@
+/// Plugin for withdrawing tokens from DAO.
+module StarcoinFramework::WithdrawPlugin {
+    use StarcoinFramework::Errors;
+    use StarcoinFramework::Option;
+    use StarcoinFramework::DAOPluginMarketplace;
+    use StarcoinFramework::DAOSpace::{Self, CapType};
+    use StarcoinFramework::Signer;
+    use StarcoinFramework::Vector;
+    use StarcoinFramework::Token;
+    use StarcoinFramework::Treasury;
+    use StarcoinFramework::Account;
+    use StarcoinFramework::InstallPluginProposalPlugin;
+
+    /// Only receiver can execute token withdraw proposal
+    const ERR_NOT_RECEIVER: u64 = 101;
+    const ERR_INVALID_SCALE_FACTOR: u64 = 102;
+
+    struct WithdrawPlugin has store, drop {}
+
+    /// WithdrawToken request.
+    struct WithdrawTokenAction<phantom TokenT> has copy, drop, store {
+        /// the receiver of withdraw tokens.
+        receiver: address,
+        /// Withdraw amount.
+        amount: u128,
+    }
+
+    /// Scale up quorum_votes for withdraw proposal.
+    /// `scale` must be in [0, 100].
+    /// The final quorum_votes = (1.0 + scale / 100) * base_quorum_votes
+    struct QuorumScale has copy, drop, store {
+        scale: u8,
+    }
+
+    public fun initialize(_sender: &signer) {
+        let witness = WithdrawPlugin {};
+
+        DAOPluginMarketplace::register_plugin<WithdrawPlugin>(
+            &witness,
+            b"0x1::WithdrawPlugin",
+            b"The plugin for withdraw token from Treasury.",
+            Option::none(),
+        );
+
+        let implement_extpoints = Vector::empty<vector<u8>>();
+        let depend_extpoints = Vector::empty<vector<u8>>();
+
+        DAOPluginMarketplace::publish_plugin_version<WithdrawPlugin>(
+            &witness,
+            b"v0.1.0",
+            *&implement_extpoints,
+            *&depend_extpoints,
+            b"inner-plugin://withdraw-plugin",
+        );
+    }
+
+    public fun required_caps(): vector<CapType> {
+        let caps = Vector::singleton(DAOSpace::proposal_cap_type());
+        Vector::push_back(&mut caps, DAOSpace::modify_config_cap_type());
+        Vector::push_back(&mut caps, DAOSpace::withdraw_token_cap_type());
+        caps
+    }
+
+    fun withdraw_limitation<DAOT: store, TokenT: store>(): u128 {

```

Comment:
This method is not used anywhere. It seems that the withdrawal amount is limited when creating a proposal